src/AppInstallerCLICore/Workflows/WorkflowBase.cpp

```diff
@@ -731,26 +732,50 @@ namespace AppInstaller::CLI::Workflow
                 if (updateAvailable || !m_onlyShowUpgrades)
                 {
                     Utility::LocIndString availableVersion, sourceName;
+                    Utility::LocIndString packageId = match.Package->GetProperty(PackageProperty::Id);
 
                     if (latestVersion)
                     {
                         if (updateAvailable)
                         {
                             availableVersion = latestVersion->GetProperty(PackageVersionProperty::Version);
-                            availableUpgradesCount++;
+                            if (context.Args.Contains(Execution::Args::Type::ListAll) || !packageIdsPrinted.count(packageId))

```

Comment:
Is there any possibility that there could be two packages with the same ID from different sources?

Reply:
There is a possibility, yes. Should the set be a pair of the package ID and source?

Reply:
I think that might prevent false positives

Reply:
yes, PackageId + SourceId will ensure a unique "winget package"

Reply:
Resolved!